### ISSUE TITLE
Only show warnings for uninitialised non-serialised fields

### ISSUE THREAD

**Issue by citizenmatt** [JetBrains/resharper-unity#1552]
The existing warning for uninitialised private fields is ReSharper reporting [CS0169](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0169). Unity disables this warning in project generation, because serialised fields don't have to be initialised - Unity will do it for you.

However, this is a project wide setting, and can hide real code issues. Removing the disabled warning from the project files will introduce warnings for existing code, as any private serialised field is likely to be uninitialised. If a project's style is to use private serialised fields, this could introduce a lot of warnings.

Unity's compilation pipeline does not use the generated C# projects, so it won't affect Unity's Console output if we remove the `<NoWarn>0169</NoWarn>` from the generated projects.

In order to catch the genuine problems with uninitialised private non-serialised fields, we should either:

* Remove `<NoWarn>0169</NoWarn>` from the generated projects and programmatically suppress the ReSharper warning on private serialised fields
* Or leave `<NoWarn>0169</NoWarn>` in the projects and add a new inspection for uninitialised private non-serialiesd fields

**Comment by Baste-RainGames** [comment]
This issue is well-known in the Unity community. It's the main reason that Roslyn introduced the [DiagnosticSuppressor API](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.diagnosticsuppressor?view=roslyn-dotnet). See the issue that lead to it being implemented [here](https://github.com/dotnet/roslyn/issues/30172).

This hasn't been implemented by Unity yet, as they [have to update their Roslyn version first](https://issuetracker.unity3d.com/issues/serializedfield-fields-produce-field-is-never-assigned-to-dot-dot-dot-warning).

So, in short, you probably shouldn't put too much work into fixing this on your side, as Unity will eventually get around to fixing it on their side.
